File: functions/alsa-lib/variants/go/1_iec958_parity_9.go
package main

func iec958_parity(data uint32) uint32 {
	data >>= 4
	parity := uint32(0)
	for data > 0 {
		parity ^= data & 1
		data >>= 1
	}
	return parity & 1
}


func main(){
    iec958_parity(0)
}

File: functions/alsa-lib/variants/c/1_iec958_parity_9.c
#include <stdint.h>

static unsigned int iec958_parity(unsigned int data)
{
	unsigned int parity = 0;
	data >>= 4;
	for (int bit = 4; bit <= 30; bit++) {
		parity ^= (data >> (bit - 4)) & 1;
	}
	return parity & 1;
}


int main(){
    iec958_parity(0);
}